sdk/iOS/src/MSManagedObjectObserver.m

```diff
@@ -7,35 +7,122 @@
 //
 
 #import "MSManagedObjectObserver.h"
+#import "MSCoreDataStore.h"
+#import "MSClient.h"
 #import <CoreData/CoreData.h>
 
 @interface MSManagedObjectObserver()
-@property (nonatomic, weak) MSClient *client;
+@property (nonatomic, strong) MSClient *client;
+@property (nonatomic, weak) NSManagedObjectContext *context;
 @end
 
 @implementation MSManagedObjectObserver
 
-- (id) initWithClient:(MSClient *)client
-{
-    return [self initWithClient:client contextsToObserver:nil];
-}
-
-- (id) initWithClient:(MSClient *)client contextsToObserver:(NSArray *)contexts
+- (instancetype) initWithClient:(MSClient *)client
 {
     self = [super init];
     if (self) {
-        _client = client;
+		// Copy so we can change the handling of how the sync table operations are handled
+        _client = [client copy];
+		
+		if ([_client.syncContext.dataSource isKindOfClass:[MSCoreDataStore class]])
+		{
+			MSCoreDataStore *dataStore = _client.syncContext.dataSource;
+			_context = dataStore.context;
+			
+			[[NSNotificationCenter defaultCenter] addObserver:self
+													 selector:@selector(handleDidSaveNotification:)
+														 name:NSManagedObjectContextDidSaveNotification
+													   object:_context];
+		}
+		
+		// Modify the handling of sync table operations for this instance of the data source
+		_client.syncContext.dataSource.handlesSyncTableOperations = FALSE;
     }
     return self;
 }
 
-- (void) registerAsObserver
+- (void)dealloc
+{
+	if (self.context != nil)
+	{
+		[[NSNotificationCenter defaultCenter] removeObserver:self
+														name:NSManagedObjectContextDidSaveNotification
+													  object:self.context];
+	}
+}
+
+/// Checks the entity has the ms_version attribute to signal it should
+/// be synchronised to the mobile service
+- (BOOL)entityHasRemoteStoreAttributes:(NSManagedObject *)managedObject
 {
-    [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(handleDidSaveNotification:)
-                                                 name:NSManagedObjectContextDidSaveNotification
-                                               object:nil];
+	return (managedObject.entity.attributesByName[StoreVersion] != nil);
```

Comment:
I didn't really intend to require all LocalStores to use ms_version long term.  Or actually require the use of version in the table, optimistic concurrency while recommended, is not technically required.

I originally was thinking we'd just ignore "ms_" tables, but I see how this lets the dev use tables for their own purposes as well.  

For prototypes this is fine, long term we may need to revisit this logic.